src/main/java/org/jenkinsci/plugins/mesos/MesosRetentionStrategy.java

```diff
@@ -86,7 +86,11 @@ private long checkInternal(MesosComputer c) {
 
       if (idleMilliseconds > MINUTES.toMillis(idleTerminationMinutes)) {
         LOGGER.info("Disconnecting idle computer " + c.getName());
-        c.getNode().terminate();
+        c.getNode().setPendingDelete(true);
```

Comment:
Oh. Didn't realize that a slave is considered idle if it's doing post build things!?